doc/releases/release-notes-2.3.rst

```diff
@@ -21,6 +21,11 @@ No security vulnerabilities received.
 API Changes
 ***********
 
+* HWINFO
+
+  * HWINFO drivers are responsible for ensuring that the ID data structure is a
```

Comment:
@sslupsky could you add a bit more info here? About the actual change in API behavior, so users know what to expect. Please also list which platforms in particular changed.

Reply:
Ok.  Clarified the behaviour and referenced the platforms changed.